### ISSUE TITLE
Header modification policy strips port number from string

### ISSUE THREAD

**Issue by kevprice83** [3scale/apicast#935]
This policy allows for the new or additional Header value to be defined as a string or liquid value.

For example: `Host: api.com:80` would be added to the policy config with `Host` & `api.com:80` as two separate values in the same policy configuration.

##### Version
- Image: registry.access.redhat.com/3scale-amp23/apicast-gateway:1.11-1

Running on Docker with following version:
- Client:
 Version:         1.13.1
 API version:     1.26
 Package version: docker-1.13.1-61.git9cb56fd.fc28.x86_64
 Go version:      go1.10.3
 Git commit:      1556cce-unsupported
 Built:           Wed Aug  1 17:21:17 2018
 OS/Arch:         linux/amd64

Server:
 Version:         1.13.1
 API version:     1.26 (minimum version 1.12)
 Package version: docker-1.13.1-61.git9cb56fd.fc28.x86_64
 Go version:      go1.10.3
 Git commit:      1556cce-unsupported
 Built:           Wed Aug  1 17:21:17 2018
 OS/Arch:         linux/amd64
 Experimental:    false

- image timestamp: `2018-09-25T20:54:25.389199Z`

##### Steps To Reproduce

1. Configure **Header modification policy** with **op**: _Create the header when not set, replace the value when set._ **header**: _Host_ **value**: _<my-api.com>:<port>_
2. `docker run` using the latest configuration for the service with above policy config.

##### Current Result
Host header is sent to the upstream as `Host: <my-api.com>`

##### Expected Result
Host header is sent to the upstream as `Host: <my-api.com>:<port>`

##### Additional Information
The expected format is compliant with the [RFC](https://tools.ietf.org/html/rfc7230#section-5.4) and is an accepted format.

[Log entry in error logs](https://gist.github.com/kevprice83/4d2d887aa6cdbb97ad281c75e9b16685)
[Example apicast config.json](https://gist.github.com/kevprice83/68b37d28af02a1230081e349cdfd0b7d)

**Comment by mikz** [comment]
@kevprice83 do I understand this correctly, that when you set header `Host` to `example.com:8080` it is sent upstream as `example.com`?

**Comment by kevprice83** [comment]
@mikz correct, this might be considered an enhancement rather than a bug of course. The RFC states the port is accepted although in my experience it's not that common.

**Comment by kevprice83** [comment]
oh  I see why it might not have been clear now, the markdown is not rendering the port correctly in my original description

**Comment by mikz** [comment]
@kevprice83 you can just configure host header in the UI and there it works just fine.
<img width="837" alt="screenshot 2018-10-17 at 16 59 04" src="https://user-images.githubusercontent.com/154571/47095681-f9bbdd00-d22d-11e8-89a6-f6897029f32a.png">

**Comment by kevprice83** [comment]
@mikz that's interesting because on the previous version I can modify the `Host` header.

Image: **registry.access.redhat.com/3scale-amp22/apicast-gateway:1.8-12**
Timestamp: **2018-08-17T08:52:30.421763Z**

With this image the `Host` header is modified using the same policy config.

**Comment by kevprice83** [comment]
@mikz it's not possible to add the port in the **host header rewrite** field until [THREESCALE-1408](https://issues.jboss.org/browse/THREESCALE-1408) is fixed. Otherwise yeah that was my first solution.

**Comment by kevprice83** [comment]
@mikz so what will the decision be here? To fix the hostname rewrite field and keep the current behaviour? Just wondering if the change in behaviour on APIcast between versions is unexpected?

**Comment by kevprice83** [comment]
@mikz what will be the decision here then? To fix the hostname rewrite field and use that or change the current behaviour in APIcast? Wondering if the change in behaviour between versions is unexpected or not?

**Comment by kevprice83** [comment]
@mikz what will be the decision here then? To fix the hostname rewrite field and use that or change the current behaviour in APIcast? Wondering if the change in behaviour between versions is unexpected or not?

**Comment by mikz** [comment]
@kevprice83 3scale/porta is open source, so fixing [THREESCALE-1408](https://issues.jboss.org/browse/THREESCALE-1408) should be a trivial task.

However, that would solve only the main endpoint of the configuration, not servers set by the upstream policy.
The change of behaviour is on hand unexpected (because we did not know about it), but also expected, because we rewrote the whole upstream subsystem to be more consistent.

If we aim to fix it in APIcast, then this would require several changes:

* ability to turn off / configure hostname rewrite in the APIcast policy and
* ability to turn off setting host header in the APIcast policy or
* changes required to make Headers policy to override Host header set by APIcast
